spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1066,9 +1066,6 @@ def _new_connection_file(self):
             cf = cf if not os.path.exists(cf) else ''
         return cf
 
-    def _shellwidget_started(self, client):
-        self.sig_shellwidget_created.emit(client.shellwidget)
-
```

Comment:
Please restore this method with the idea of calling it below. I think the code is easier to follow if we have `_shellwidget_created` and its opposite `_shellwidget_created`.